test/database/download_test.go

```diff
@@ -0,0 +1,29 @@
+package database
+
+import (
+	"context"
+	"testing"
+
+	"github.com/SkynetLabs/skynet-accounts/database"
+	"github.com/SkynetLabs/skynet-accounts/test"
+)
+
+// TestDownloadCreateAnon ensures that UploadCreate can create anonymous downloads.
+func TestDownloadCreateAnon(t *testing.T) {
+	ctx := context.Background()
+	dbName := test.DBNameForTest(t.Name())
+	db, err := database.NewCustomDB(ctx, dbName, test.DBTestCredentials(), nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	sl := test.RandomSkylink()
+	skylink, err := db.Skylink(ctx, sl)
+	if err != nil {
+		t.Fatal(err)
+	}
+	// Register an anonymous download.
+	err = db.DownloadCreate(ctx, nil, *skylink, 123)
```

Comment:
this is not really testing anything right, if the impl. of `DownloadCreate` was just `return nil` this test would succeed, should query the database and assert it got inserted, and the user id was the anon user

Reply:
file deleted